teammapper-frontend/src/app/guards/toast.guard.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import {
+  CanActivate,
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot,
+  Router,
+} from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ToastGuard implements CanActivate {
+  constructor(private router: Router, private toastrService: ToastrService) {}
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Promise<boolean> | boolean {
+    const showToast = next.queryParamMap.get('showToast');
+    const toastMessage = next.queryParamMap.get('toastMessage');
+
+    if (showToast === 'true' && toastMessage) {
```

Comment:
instead of true/false as strings, maybe better to use 1/0?

Reply:
wouldnt be just the availability of a toastMessage enough? Or what's the case where I am passing a toast message without the showToast flag?

Reply:
Yeah good point about the toastMessage actually :D